File: New Project/calc.vb
Module Module1
 
    Public Sub Main()
        Dim n As Integer = Console.ReadLine(), c As Char = Console.ReadLine(), m As Integer = System.Console.ReadLine(), answer As String, i As Integer
        Do Until n = 0 And m = 0
            answer = ""
            Select Case c
                Case "+"
                    answer = n + m
                Case "-"
                    answer = n - m
                Case "*"
                    answer = n * m
                Case "/"
                    If m Then answer = n / m Else answer = "Error dividing by 0\n"
                Case "^"
                    answer = 1
                    If m < 0 Then
                        answer = "-ve power is not supported yet\n"
                    Else
                        For i = 0 To m
                            answer *= n
                        Next
                    End If
                Case Else
                    answer = "This operation is invalid or not supported yet\n"
            End Select
            Console.WriteLine(answer)
            n = Console.ReadLine()
            c = Console.ReadLine()
            m = Console.ReadLine()
        Loop
    End Sub
 
End Module

File: New Project/calc.cpp
#include <bits/stdc++.h>
 
using namespace std;
 
int main()
{
    int n, m, answer;
    string error_type;
    char c;
    while(cin >> n >> c >> m && (n || m)){
        error_type = "";
        switch(c){
        case '+':
            answer = n + m;break;
        case '-':
            answer = n - m;break;
        case '*':
            answer = n * m;break;
        case '/':
            if(m)answer = n / m;
            else error_type = "Error dividing by 0\n";
            break;
        case '^':
            answer = 1;
            if(m < 0)error_type = "-ve power is not supported yet\n";
            else
                for(int i = 0; i < m; i++)answer *= 2;
            break;
        default:
            error_type = "This operation is invalid or not supported yet\n";
        }
        if(!error_type.empty())printf(&error_type[0]);
        else printf("%d\n", answer);
    }
    return 0;
}
